src/modules/ekf2/EKF/mag_control.cpp

```diff
@@ -250,63 +254,94 @@ bool Ekf::haglYawResetReq()
 	return false;
 }
 
-bool Ekf::magReset()
+bool Ekf::resetMagStates(const Vector3f &mag)
 {
 	// prevent a reset being performed more than once on the same frame
 	if ((_flt_mag_align_start_time == _time_delayed_us)
 	    || (_control_status_prev.flags.yaw_align != _control_status.flags.yaw_align)) {
 		return false;
 	}
 
+	bool reset_heading = true;
```

Comment:
I guess this can become `false` in the next PR?

Reply:
In https://github.com/PX4/PX4-Autopilot/pull/21212 it's passed in so that we can optionally reset just the mag states if we're not in mag_hdg or mag_3D.